src/NuGet.Core/NuGet.Packaging/PackagesConfigWriter.cs

```diff
@@ -206,22 +236,26 @@ public void UpdateOrAddPackageEntry(XDocument originalConfig, PackageReference n
             var originalPackagesNode = originalConfig.Element(XName.Get(PackagesConfig.PackagesNodeName));
 
             XElement matchingIdNode;
-            if (PackagesConfig.HasAttributeValue(
-                originalPackagesNode,
-                PackagesConfig.IdAttributeName,
-                newEntry.PackageIdentity.Id,
-                out matchingIdNode))
-            {
-                // Find the old entry and update it based on the new entry
-                var packagesNode = _xDocument.Element(XName.Get(PackagesConfig.PackagesNodeName));
-                var newPackageNode = ReplacePackageAttributes(matchingIdNode, newEntry);
-                packagesNode.Add(newPackageNode);
-                SortPackageNodes(packagesNode);
-            }
-            else
+
+            if (originalPackagesNode != null)
             {
-                // There was no existing entry, add a new one
-                AddPackageEntry(newEntry);
+                if (PackagesConfig.HasAttributeValue(
+                    originalPackagesNode,
+                    PackagesConfig.IdAttributeName,
+                    newEntry.PackageIdentity.Id,
+                    out matchingIdNode))
+                {
+                    // Find the old entry and update it based on the new entry
+                    var packagesNode = _xDocument.Element(XName.Get(PackagesConfig.PackagesNodeName));
+                    var newPackageNode = ReplacePackageAttributes(matchingIdNode, newEntry);
+                    packagesNode.Add(newPackageNode);
+                    SortPackageNodes(packagesNode);
+                }
+                else
+                {
+                    // There was no existing entry, add a new one
+                    AddPackageEntry(newEntry);
```

Comment:
the new entry isn't added here if the original config is null, it should be added either way